test/rules/avoid_function_literals_in_foreach_calls.dart

```diff
@@ -17,4 +21,17 @@ void main() {
   people.forEach((person) => print(person)); // LINT
 
   people.forEach(print); // OK
+
+  people
+      .where((person) => person != null)
+      .map((person) => '$person!')
+      .forEach((person) => print(person)); // OK
+
+  Person()
+      .children
+      .firstWhere((person) => person != null)
+      .children
+      .forEach((person) => print(person)); // OK
+
+  Person().children.first.children.forEach((person) => print(person)); // OK
```

Comment:
Interesting.  I thought the proposal was limited to method-chains.

IOW, I would probably still prefer this one:

```dart
for (var person in Person().children.first.children) {
  print(person);
}
```

(which feels different than the `.where(..).map(..)` case...)

/cc @sjmcdowall @munificent

Reply:
I agree. It seems unnecessary and possibly undesirable to me to special case getters, but I'll defer to Bob on that question.

Reply:
So I initially went for only method chaining which mitigated the where/map issues, but then I felt a bit in the middle for long field/getter chaining... I'm OK either way so I'll defer to you guys.

Reply:
I don't think I have a useful opinion either way here. I think it's good for the linter to *not* yell at uses of `forEach()` at the end of method chains, but I don't have a strong sense of when I think it *should* complain.

Reply:
OK since the initial issue/request was mainly focus on loosen the lint on method calls such as where/map maybe we could start limiting to only method-chains? I can update to that if we all agree

Reply:
FWIW -- I agree @dramos07  :)

Reply:
That sounds reasonable to me. If the getter chains are causing a real issue, then we can always revisit (though changing it later will be a breaking change).

Reply:
Limiting it to method chains SGTM.  Thanks!

Reply:
Updated the code to only account for method chained calls, thanks everyone!!